src/hazelweb/gui/KeyComboAction.re

```diff
@@ -1,56 +1,58 @@
-  ]
-  |> List.to_seq
-  |> Hashtbl.of_seq;
+let get_model_action =
+    (cursor_info: CursorInfo.t, kc: HazelKeyCombos.t, is_mac: bool)
+    : option(ModelAction.t) =>
+  switch (kc, cursor_info, is_mac) {
+  | (Escape, _, _) => None
+  | (Backspace, _, _) => Some(EditAction(Backspace))
+  | (Delete, _, _) => Some(EditAction(Delete))
+  | (ShiftTab, _, _) => Some(EditAction(MoveToPrevHole))
+  | (Tab, _, _) => Some(EditAction(MoveToNextHole))
+  | (GT, {CursorInfo.typed: OnType, _}, _) =>
+    Some(EditAction(Construct(SOp(SArrow))))
+  | (GT, _, _) => Some(EditAction(Construct(SOp(SGreaterThan))))
+  | (Ampersand, _, _) => Some(EditAction(Construct(SOp(SAnd))))
+  | (VBar, _, _) => Some(EditAction(Construct(SOp(SOr))))
+  | (LeftParen, _, _) => Some(EditAction(Construct(SParenthesized)))
+  | (Colon, _, _) => Some(EditAction(Construct(SAnn)))
+  | (Equals, _, _) => Some(EditAction(Construct(SOp(SEquals))))
+  | (Enter, _, _) => Some(EditAction(Construct(SLine)))
+  | (Shift_Enter, _, _) => Some(EditAction(Construct(SCommentLine)))
+  | (Backslash, _, _) => Some(EditAction(Construct(SLam)))
+  | (Plus, _, _) => Some(EditAction(Construct(SOp(SPlus))))
+  | (Minus, _, _) => Some(EditAction(Construct(SOp(SMinus))))
+  | (Asterisk, _, _) => Some(EditAction(Construct(SOp(STimes))))
+  | (Slash, _, _) => Some(EditAction(Construct(SOp(SDivide))))
+  | (LT, _, _) => Some(EditAction(Construct(SOp(SLessThan))))
+  | (Space, {CursorInfo.cursor_term: Line(_, CommentLine(_)), _}, _) =>
+    Some(EditAction(Construct(SChar(" "))))
+  | (Space, _, _) => Some(EditAction(Construct(SOp(SSpace))))
+  | (Comma, _, _) => Some(EditAction(Construct(SOp(SComma))))
+  | (LeftBracket, {CursorInfo.typed: OnType, _}, _) =>
+    Some(EditAction(Construct(SList)))
+  | (LeftBracket, _, _) => Some(EditAction(Construct(SListNil)))
+  | (Semicolon, _, _) => Some(EditAction(Construct(SOp(SCons))))
+  | (Alt_L, _, _) => Some(EditAction(Construct(SInj(L))))
+  | (Alt_R, _, _) => Some(EditAction(Construct(SInj(R))))
+  | (Alt_C, _, _) => Some(EditAction(Construct(SCase)))
+  | (Pound, _, _) => Some(EditAction(Construct(SCommentLine)))
+  | (Ctrl_Z, _, true) => None
+  | (Ctrl_Z, _, false) => Some(Undo)
+  | (Ctrl_Shift_Z, _, true) => None
+  | (Ctrl_Shift_Z, _, false) => Some(Redo)
+  | (Ctrl_Alt_I, _, _) => Some(EditAction(SwapUp))
+  | (Ctrl_Alt_K, _, _) => Some(EditAction(SwapDown))
+  | (Ctrl_Alt_J, _, _) => Some(EditAction(SwapLeft))
+  | (Ctrl_Alt_L, _, _) => Some(EditAction(SwapRight))
+  | (Meta_Z, _, true) => Some(Undo)
+  | (Meta_Z, _, false) => None
+  | (Meta_Shift_Z, _, true) => Some(Redo)
+  | (Meta_Shift_Z, _, false) => None
+  };
 
-let get = (cursor_info: CursorInfo.t, kc: HazelKeyCombos.t): Action.t => {
-  let action_of_ci = Hashtbl.find(table, kc);
-  action_of_ci(cursor_info);
-};
+let get_action =
+    (cursor_info: CursorInfo.t, kc: HazelKeyCombos.t): option(Action.t) =>
+  // is_mac shouldn't matter here, so we'll just have it be false
```

Comment:
this is just an accident of the current set of actions though -- lets set it up to propagate that information correctly.

Reply:
In commit [`78dd45c`](https://github.com/hazelgrove/hazel/commit/78dd45c37c61c4cd68205121e8167e821e8e910a) I made it so that instead of `is_mac` being a `bool`, it's an `option(bool)` so that it can account for applications of the function where it's not known whether the user is on a Mac, such as in `ActionPanel.re`.

Is this what you were expecting?

Reply:
Well I was suggesting that some edit actions might actually need that information in the future, so lets set it up so that is being passed in correctly (which might require changing something upstream of this function to propagate that correctly.)

Reply:
Okay, I think that makes sense. I made a new commit ([`fb7261`](https://github.com/hazelgrove/hazel/commit/fb7261a5242b0232a73e7316a04558e1df81fec2)) that does this.